app/models/concerns/unparsed_run.rb

```diff
@@ -4,60 +4,10 @@ module UnparsedRun
   extend ActiveSupport::Concern
 
   included do
-    def parses?(fast: true, convert: false)
-      parse(fast: fast, convert: convert).present? && timer.present?
-    end
-
     def parsed?
       parsed_at.present?
     end
 
-    def parse(fast: true, convert: false)
-      return @parse_cache[fast] if @parse_cache.try(:[], fast).present?
-      return @parse_cache[false] if @parse_cache.try(:[], false).present?
-      return @convert_cache if @convert_cache.present?
-
-      unless convert
-        if fast
-          unless parsed?
-            parse_into_db
-            return nil unless parsed?
-          end
-
-          return {
-            id: id36,
-            timer: program,
-            attempts: attempts,
-            srdc_id: srdc_id,
-            duration: (realtime_duration_ms || 0) / 1000,
-            sum_of_best: (realtime_sum_of_best_ms || 0) / 1000,
-            splits: segments
-          }
-        end
-      end
-
-      # Convert flow below
-
-      parse_result = Parser.parse(file, fast: fast)
-      return {} if parse_result.nil?
-
-      parse_result[:timer] = Run.program_from_attribute(:to_s, parse_result[:program]).to_sym
-      assign_attributes(
-        program: parse_result[:timer],
-        attempts: parse_result[:attempts],
-        srdc_id: srdc_id || parse_result[:srdc_id].presence,
-        realtime_duration_s: parse_result[:splits].map(&:realtime_duration).sum.to_f,
-        realtime_sum_of_best_s: parse_result[:splits].map.all? do |split|
-          split.realtime_best.present?
-        end && parse_result[:splits].map(&:realtime_best).sum.to_f
-      )
-
-      @convert_cache = parse_result
-      @segments_cache = parse_result[:splits]
-
-      parse_result
-    end
-
```

Comment:
oh man that feels so good. this method has been around basically since day 1

Reply:
It's always nice to go back and clean up all the old code :)